ui/src/trace_processor/http_rpc_engine.ts

```diff
@@ -86,11 +88,21 @@ export class HttpRpcEngine extends EngineBase {
   }
 
   private onWebsocketMessage(e: MessageEvent) {
-    assertExists(e.data as Blob)
-      .arrayBuffer()
-      .then((buf) => {
-        super.onRpcResponseBytes(new Uint8Array(buf));
-      });
+    const blob = assertExists(e.data as Blob);
+    this.queue.push(blob);
+    this.processQueue();
+  }
+
+  private async processQueue() {
+    if (this.isProcessing) return;
+    this.isProcessing = true;
+    while (this.queue.length > 0) {
+      const blob = this.queue.shift();
+      if (!blob) continue;
```

Comment:
Is there any case when the blob would be falsy?
if the queue length > 0, blob must be defined right?
So this should be a `const blob = assertExists(this.queue.shift())` rather than being defensive